Dockerfile

```diff
@@ -0,0 +1,119 @@
+RUN touch ./src/main.rs
+# Build exa
+RUN cargo build
+
+# We use Ubuntu instead of Debian because the image comes with two-way
+# shared folder support by default.
+# This image is based from Ubuntu
+FROM ${BASE_RUST} AS base
+
+# Install the dependencies needed for exa to build
+RUN --mount=type=cache,target=/var/cache/apt apt-get update -qq
+RUN --mount=type=cache,target=/var/cache/apt apt-get install -qq -o=Dpkg::Use-Pty=0 \
+    git gcc curl attr libgit2-dev zip \
+    fish zsh bash bash-completion
+
+# Install kcov for test coverage
+# This doesn’t run coverage over the xtests so it’s less useful for now
+COPY --from=cargo-kcov /usr/local/cargo/bin/cargo-kcov /usr/bin/cargo-kcov
+
+RUN --mount=type=cache,target=/var/cache/apt apt-get install -y \
+    cmake g++ pkg-config \
+    libcurl4-openssl-dev libdw-dev binutils-dev libiberty-dev
+
+RUN ln -s $(which python3) /usr/bin/python
+RUN cargo kcov --print-install-kcov-sh | sh
+
+# Create a variety of misc scripts.
+
+RUN ln -sf /vagrant/devtools/dev-run-debug.sh /usr/bin/exa
+RUN ln -sf /vagrant/devtools/dev-run-release.sh /usr/bin/rexa
+
+RUN echo -e "#!/bin/sh\ncargo build --manifest-path /vagrant/Cargo.toml \\$@" > /usr/bin/build-exa
+RUN ln -sf /usr/bin/build-exa /usr/bin/b
+
+RUN echo -e "#!/bin/sh\ncargo test --manifest-path /vagrant/Cargo.toml \\$@ -- --quiet" > /usr/bin/test-exa
+RUN ln -sf /usr/bin/test-exa /usr/bin/t
+
+RUN echo -e "#!/bin/sh\n/vagrant/xtests/run.sh" > /usr/bin/run-xtests
+RUN ln -sf /usr/bin/run-xtests /usr/bin/x
+
+RUN echo -e "#!/bin/sh\nbuild-exa && test-exa && run-xtests" > /usr/bin/compile-exa
+RUN ln -sf /usr/bin/compile-exa /usr/bin/c
+
+ADD --chmod=+x devtools/dev-package-for-linux.sh /vagrant/devtools/dev-package-for-linux.sh
+RUN echo -e "#!/bin/sh\nbash /vagrant/devtools/dev-package-for-linux.sh \\$@" > /usr/bin/package-exa
+RUN echo -e "#!/bin/sh\ncat /etc/motd" > /usr/bin/halp
+
+# Configure the welcoming text that gets shown:
+
+# Capture the help text so it gets displayed first
+RUN rm -f /etc/update-motd.d/*
+ADD --chmod=+x devtools/dev-help.sh /vagrant/devtools/dev-help.sh
+RUN bash /vagrant/devtools/dev-help.sh > /etc/motd
+
+# Tell bash to execute a bunch of stuff when a session starts
+RUN echo "source /vagrant/devtools/dev-bash.sh" > ${HOME}/.bash_profile
+
+# Link the completion files so they’re “installed”:
+
+# bash
+RUN ln -s /vagrant/completions/bash/exa /etc/bash_completion.d/exa
+RUN ln -s /vagrant/completions/bash/exa /usr/share/bash-completion/completions/exa
+
+# zsh
+RUN ln -s /vagrant/completions/zsh/_exa /usr/share/zsh/vendor-completions/_exa
+
+# fish
+RUN ln -s /vagrant/completions/fish/exa.fish /usr/share/fish/completions/exa.fish
+
+
+# Copy the source code into the image
+COPY --chmod=+x . /vagrant/
+
+# Make sudo dummy replacement
+# This is needed for some tests that use sudo
+RUN echo -e '#!/bin/sh\n"$@"' > /usr/bin/sudo
+RUN chmod +x /usr/bin/sudo
+
+RUN bash /vagrant/devtools/dev-set-up-environment.sh
```

Comment:
<details>
<summary>Ignore this</summary>

This also has some odd locale logic at the end. I'm wondering if that's necessary?

In [`docker-mailserver` image](https://github.com/docker-mailserver/docker-mailserver/blob/f28fce9cc432f1f447bd963d9e54e44bcf2c27dd/Dockerfile#L275-L280) we have a similar locale section:

```Dockerfile
RUN <<EOF
  rm -rf /usr/share/locale/*
  rm -rf /usr/share/man/*
  rm -rf /usr/share/doc/*
  update-locale
EOF
```

For exa, I tracked it down to a specific test for different locales, so ignore this feedback 😅 

https://github.com/ogham/exa/commit/115315a03cefbdc6ec0da7ab174113a5950434f3

</details>